src/Engine/Image.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Engine;
+
+use Imagine\Filter\Basic\Crop;
+use Imagine\Image\Box;
+use Imagine\Image\Point;
+
+class Image
+{
+
+    /** @var string Original URL of image. */
+    private $imageUrl;
+
+    /** @var int[] Array of floats with keys: `x`, `y`, `width`, and `height`. */
+    private $crop;
+
+    /** @var string Image data. */
+    private $data;
+
+    /** @var int Image size in bytes. */
+    private $size;
+
+    /**
+     * @param int[] $crop
+     */
+    public function __construct(string $imageUrl, array $crop = [])
+    {
+        $this->imageUrl = $imageUrl;
+        $this->crop = $crop;
+    }
+
+    public function geturl(): string
+    {
+        return $this->imageUrl;
+    }
+
+    public function needsCropping(): bool
+    {
+        return isset($this->crop['width']) && $this->crop['width'] > 0
+            && isset($this->crop['height']) && $this->crop['height'] > 0;
```

Comment:
Should this be an OR? I.e.

```
return ( isset($this->crop['width']) && $this->crop['width'] > 0 )
            || ( isset($this->crop['height']) && $this->crop['height'] > 0 );
```

This would only make a difference if width or height is 0, in which case we're currently using the whole image. It shouldn't matter for ordinary usage anyway.

Reply:
Oh, I didn't see that error. But yeah, it doesn't feel necessary to complicate this. Thus, it's fine to leave the code as it currently is.